src/cli/args.rs

```diff
@@ -41,6 +41,10 @@ pub struct CliArgs {
     #[arg(short, long, global = true)]
     pub format: Option<OsString>,
 
+    /// decompress or list with password
+    #[arg(short = 'p', long = "password", global = true)]
+    pub password: Option<String>,
```

Comment:
Ideally, here we'd use OsString to support non-UTF-8 passwords (that's why zip and rar ask for `&[u8]` and not `&str`)

Idk where Clap documents that it supports OsString but it does :smile:.

(After the change you can use `args.password.as_ref().map(|string| string.as_bytes()),` in the two places it will complain.)

---

This suggestion goes together with the other suggestion of using `Option<&[u8]>` in list and decompression function signatures instead of `impl AsRef<[u8]>`.

Reply:
Argh this suggestion also seems wrong, my bad.